src/quick-lint-js/parse.h

```diff
@@ -3037,7 +3037,6 @@ class parser {
 
         switch (this->peek().type) {
         // let switch = 3;  // Invalid.
-        case token_type::end_of_file:
```

Comment:
Blocking: This deletion changes the behavior of `let var` or `var var` at the end of a file, as the failing test points out. We should make `var var` (at end of file) continue to report E124.